add docs for google map field (#377)
Laravel-Backpack/docs: 5.x/crud-fields.md

<hr>

<a name="google-map"></a>

> ### google_map

Reviewer: I don't think this should be under FREE fields 👀 It's a PRO field, isn't it?

> Uses the Google Places API to generate a Map where user can navigate and select a position to store `Latitude` (**lat**), `Longitude` (**lng**) and the address string (**formatted_address**). By default it saves a string in dabase with `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`. 

Reviewer: Sorry but no. This is NOT what the field does. Why are we writing from a technical perspective here? We should write from a USER perspective.

```suggestion
Shows a map and allows the user to navigate and select a position on that map (using the Google Places API). The field stores the latitude, longitude and the address string as a JSON in the database ( eg. `{lat: 123, lng: 456, formatted_address: 'Lisbon, Portugal'}`). If you want to save the info in separate db columns, continue reading below.
```

> **Note**: If you want two save in separate fields continue reading below.

Reviewer: ```suggestion
```

```php
'google_places' => [
    'key' => 'the-key-you-got-from-google-places'
],
```

### SAVING IN MULTIPLE INPUTS
There are cases where you rather save the information on separate inputs in the database. 
In that scenario you should use [Laravel mutators and accessors](https://laravel.com/docs/9.x/eloquent-mutators) .

Using the same field as previously shown (**field name is `location`**), and having `latitude`, `longitude`, `full_address` as the database fields, we can save and retrieve them with the following example:
```php

//add all the fields to model fillable property, including the one that we are not going to save (location in the example)
$fillable = ['location', 'latitude', 'longitude', 'full_address'];

// 
protected function location(): \Illuminate\Database\Eloquent\Casts\Attribute
{
    return \Illuminate\Database\Eloquent\Casts\Attribute::make(
        get: fn ($value, $attributes) => json_encode([
            'lat' => $attributes['lat'], 
            'lng' => $attributes['lng'], 
            'formatted_address' => $attributes['full_address'] ?? ''

>             ],
>              JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR),

Reviewer: This looks weird to me on a separate line
```suggestion
            ], JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_THROW_ON_ERROR),
```

        set: function($value) {
            $location = json_decode($value);
            return [
                'lat' => $location->lat,
                'lng' => $location->lng,
                'full_address' => $location->formatted_address ?? ''
            ];

>         }

Reviewer: If the getter is using arrow functions, shouldn't the setter too? Let's choose one and stick with it.

    );
}

```

> Input preview:

Reviewer: Let's upload an image for it here, inside the Github conversation. 
Then we can take that URL and introduce it in the markdown 😉
